infrastructure/general/setup-secure-wss.md

```diff
@@ -0,0 +1,126 @@
+
+### Obtaining an SSL Certificate
+
+One easy way to get a free SSL certificate can be achieved by following the LetsEncrypt instructions
+([nginx](https://certbot.eff.org/instructions?ws=nginx&os=ubuntufocal){target=_blank}/[apache](https://certbot.eff.org/instructions?ws=apache&os=ubuntufocal){target=_blank}).
+This will auto-generate an SSL certificate and include it in your configuration.
+
+Alternatively, you can generate a self-signed certificate and rely on the raw IP address of your
+node when connecting to it. This is not preferable since you will have to whitelist the certificate
+to access it from a browser.
+
+```bash
+sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/ssl/private/selfsigned.key -out /etc/ssl/certs/selfsigned.crt
+sudo openssl dhparam -out /etc/ssl/certs/dhparam.pem 2048
+```
+
+## Installing a Proxy Server
+
+There are a lot of different implementations of a WebSocket proxy, some of the more widely used are
+[nginx](https://www.nginx.com/){target=_blank} and [apache2](https://httpd.apache.org/){target=_blank}, for which configuration
+examples provided below.
+
+### Nginx
+
+```bash
+apt install nginx
+```
+
+In an SSL-enabled virtual host add:
+
+```conf
+server {
+  (...)
+  location / {
+    proxy_buffers 16 4k;
+    proxy_buffer_size 2k;
+    proxy_pass http://localhost:9944;
+    proxy_http_version 1.1;
+    proxy_set_header Upgrade $http_upgrade;
+    proxy_set_header Connection "Upgrade";
+    proxy_set_header Host $host;
+   }
+}
+```
+
+Optionally some form of rate limiting can be introduced:
+
+```conf
+http {
+  limit_req_zone  "$http_x_forwarded_for" zone=zone:10m rate=2r/s;
+  (...)
+}
+
+location / {
+  limit_req zone=zone burst=5;
+  (...)
+}
+```
+
+### Apache2
+
+You can run it in different modes such as prefork, worker, or event. In this example, we use
+[event](https://httpd.apache.org/docs/2.4/mod/event.html){target=_blank} which works well on higher load
+environments but other modes are also useful given the requirements.
+
+```bash
+apt install apache2
+a2dismod mpm_prefork
+a2enmod mpm_event proxy proxy_html proxy_http proxy_wstunnel rewrite ssl
+```
+
+The [mod_proxy_wstunnel](https://httpd.apache.org/docs/2.4/mod/mod_proxy_wstunnel.html){target=_blank} provides
+_support for the tunneling of web socket connections to a backend websockets server. The connection
+is automatically upgraded to a WebSocket connection_. In an SSL-enabled virtualhost add:
+
+```apacheconf
+(...)
+SSLProxyEngine on
+ProxyRequests off
+
+ProxyPass / ws://localhost:9944
+ProxyPassReverse / ws://localhost:9944
+```
+
+Older versions of mod_proxy_wstunnel do not upgrade the connection automatically and will need the
```

Comment:
```suggestion
Older versions of mod_proxy_wstunnel don't upgrade the connection automatically and will need the
```
We prefer the use of contractions for 'not' combinations. Readers are more likely to see the 'not' indicator if it is written "don't" or "won't" rather than as two words they might miss one if skimming, etc.